### PR TITLE
feat: add logstash support

### PR DESCRIPTION
**Pull request by stakach**
need to update rubber-soul before I can update rest-api due to the action-controller big version bump

### CODE REVIEW

File: src/constants.cr [hunk 1]
```diff
@@ -5,8 +5,31 @@ module RubberSoul
...
 
-  Log         = ::Log.for(APP_NAME)
-  LOG_BACKEND = ActionController.default_backend
+  Log           = ::Log.for(APP_NAME)
+  LOG_STDOUT    = ActionController.default_backend
+  LOGSTASH_HOST = ENV["LOGSTASH_HOST"]?
+  LOGSTASH_PORT = ENV["LOGSTASH_PORT"]?
+
+  def self.log_backend
+    if logstash_host = LOGSTASH_HOST
```
Comment (caspiano) [inline src/constants.cr:14]:
```suggestion
    if !(logstash_host = LOGSTASH_HOST.presence).nil?
      logstash_port = LOGSTAH_PORT.try &.to_i? || abort("LOGSTASH_PORT is either malformed or not present in environment')

File: src/constants.cr [hunk 2]
```diff
@@ -5,8 +5,31 @@ module RubberSoul
...
+      # Logstash UDP Input
+      logstash = UDPSocket.new
+      logstash.connect logstash_host, LOGSTASH_PORT.not_nil!.to_i
```
Comment (caspiano) [inline src/constants.cr:17]:
```suggestion
      logstash.connect logstash_host, logstash_port 
```

File: src/constants.cr [hunk 3]
```diff
@@ -5,8 +5,31 @@ module RubberSoul
...
+      logstash.sync = false
+
+      # debug at the broadcast backend level, however this will be filtered
+      # by the bindings
+      backend = ::Log::BroadcastBackend.new
+      backend.append(LOG_STDOUT, :debug)
+      backend.append(ActionController.default_backend(
+        io: logstash,
+        formatter: ActionController.json_formatter
+      ), :debug)
```
Comment (caspiano) [inline src/constants.cr:27]: lowest debug level is `trace`

```suggestion
      backend.append(LOG_STDOUT, :trace)
      backend.append(ActionController.default_backend(
        io: logstash,
        formatter: ActionController.json_formatter
      ), :trace)
```

**Review by viv-4 — APPROVED** [review]

**Comment by caspiano** [comment]
Also need to change `RubberSoul::LOG_BACKEND` in `spec/spec_config.cr` to `RubberSoul::LOG_STDOUT`

**Pull request merged**